### PR TITLE
Cleanup ReCaptchaService after changing the registration type

### PR DESCRIPTION
**Pull request by MikeAlhayek**
@jtkech @sebastienros

### CODE REVIEW

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaService.cs [hunk 1]
```diff
@@ -124,8 +125,10 @@ private async Task<bool> VerifyAsync(string responseToken)
                     { "response", responseToken }
                 });
 
-                _httpClient ??= _httpClientFactory.CreateClient(nameof(ReCaptchaService));
-                var response = await _httpClient.PostAsync($"{_reCaptchaSettings.ReCaptchaApiUri.TrimEnd('/')}/siteverify", content);
+                var httpClient = _httpClientFactory.CreateClient(nameof(ReCaptchaService));
+                _verifyHost ??= $"{_reCaptchaSettings.ReCaptchaApiUri.TrimEnd('/')}/siteverify";
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaService.cs:129]: move to the constructor, make field readonly

Comment (jtkech) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaService.cs:129]: Yes

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaService.cs]: @sebastienros done. Is there a reason for this request? what I had did the same thing except that we don't initialize until needed.

Comment (sebastienros) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaService.cs]: it doesn't need to be tested for null each time.

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Services/ReCaptchaService.cs]: okay. let me know if you want anything else changed or let's merge it
